src/buffer/cc_dbuf.c

```diff
@@ -136,13 +139,11 @@ dbuf_shrink(struct buf **buf)
     }
 
     if (nsize != buf_size(*buf)) {
-        status = _dbuf_resize(buf, nsize);
-        if (status == CC_OK) {
-            INCR(dbuf_metrics, dbuf_shrink);
-        } else {
-            INCR(dbuf_metrics, dbuf_shrink_ex);
-        }
-    }
+        /* should never fail, since we are shrinking */
+        rstatus_i status = _dbuf_resize(buf, nsize);
+        ASSERT(status == CC_OK);
+        (void)status;
```

Comment:
remove this line?

Reply:
actually, i think this assumption i made is incorrect, and `realloc` doesn't guarantee success while trimming, and it depends on the implementation. so I guess the best thing would be to return a status here, but the original buffer will still be valid.